src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +25,122 @@ interface GetCourseDetailsResponse {
-
-    if (filteredCourseDetailsList.length > 0) {
-      return filteredCourseDetailsList[0];
-    }
-
-    return null;
-  }
-
-  /*------------------------------Functions to toggle item state in state maps----------------------------------*/
-  /**
-   * Change the state of a specific course.
-   */
-  toggleCourseState(courseDetails: CourseDetails, designatedState?: boolean): void {
-    const state: boolean = designatedState === undefined
-        ? !courseDetails.isChecked
-        : designatedState;
-
-    courseDetails.isChecked = state;
 
-    if (!this.isCourseDetailsFetched(courseDetails.id)) {
-      this.courseDetailsList.push(courseDetails);
-    }
-
-    const course: Course | null = this.getCourse(courseDetails.id);
-    if (course) {
-      course.isChecked = state;
-    }
-    courseDetails.sections.forEach((section: SectionDetails) => {
-      this.toggleSectionState(section, state);
-    });
-  }
-
-  /**
-   * Change the state of a specific section.
-   */
-  toggleSectionState(section: SectionDetails, designatedState?: boolean): void {
-    const state: boolean = designatedState === undefined
-        ? !section.isChecked
-        : designatedState;
-
-    section.isChecked = state;
-
-    if (!this.isSectionDetailsFetched(section)) {
-      this.sectionDetailsList.push(section);
-    }
-
-    section.teams.forEach((team: TeamDetails) => {
-      this.toggleTeamState(team, state);
-    });
-  }
-
-  /**
-   * Change the state of a specific team.
-   */
-  toggleTeamState(team: TeamDetails, designatedState?: boolean): void {
-    const state: boolean = designatedState === undefined
-        ? !team.isChecked
-        : designatedState;
-
-    team.isChecked = state;
-
-    if (!this.isTeamDetailsFetched(team)) {
-      this.teamDetailsList.push(team);
-    }
-
-    team.students.forEach((student: StudentDetails) => {
-      this.toggleStudentState(student, state);
-    });
-  }
-
-  /**
-   * Change the state of a specific student.
-   */
-  toggleStudentState(student: StudentDetails, designatedState?: boolean): void {
-    const state: boolean = designatedState === undefined
-        ? !student.isChecked
-        : designatedState;
-
-    student.isChecked = state;
-
-    if (!this.isStudentDetailsFetched(student)) {
-      this.studentDetailsList.push(student);
+      this.viewStudent[index] = true;
     }
   }
 
-  /*------------------------------Functions to get item state from state maps-----------------------------------*/
   /**
-   * Get the list of students which need to be hidden on the StudentTable display.
-   */
-  getStudentsToHide(courseId: string): string[] {
-    return this.studentDetailsList
-        .filter((student: StudentDetails) => student.courseId === courseId && !student.isChecked)
-        .map((student: StudentDetails) => student.email);
-  }
+   * Loads students of a specified course.
+   */
+  loadStudents(courseId: string): void {
```

Comment:
I am not sure, but do we need to make it `private`?

Reply:
Looking at out other front end components, I dont think we have enforced this.